src/transport_udp.cpp

```diff
@@ -25,30 +25,30 @@ UDPTransport::~UDPTransport()
   // Stop threads
   stop = true;
 
-  // Push empty data to allow write queue to not block
-  connData cd = { .contextId = 0, .mStreamId = 0, .data = {} };
-  fd_write_queue.push(cd);
+  // Clear threads from the queue
+  fd_write_queue.stopWaiting();
 
   // Close socket fd
-  ::close(fd);
+  if (fd >= 0)
+    ::close(fd);
 
   logger.log(LogLevel::info, "Closing transport threads");
-  for (auto thr : running_threads) {
-    thr->join();
+  for (auto &thread : running_threads) {
```

Comment:
This for preference?

Reply:
vectors are almost always preferred over lists and pointers are almost always never preferred.  A list of pointers is a bad thing to use.